module/VuFind/src/VuFind/Db/Service/UserListService.php

```diff
@@ -205,12 +245,34 @@ public function getUserListsByTagAndId(
      */
     public function getUserListsByUser(UserEntityInterface|int $userOrId): array
     {
-        $userId = $userOrId instanceof UserEntityInterface ? $userOrId->getId() : $userOrId;
-        $callback = function ($select) use ($userId) {
-            $select->where->equalTo('user_id', $userId);
-            $select->order(['title']);
-        };
-        return iterator_to_array($this->getDbTable('UserList')->select($callback));
+        $dql = 'SELECT ul '
+            . 'FROM ' . $this->getEntityClass(UserList::class) . ' ul '
+            . 'WHERE ul.user = :user '
+            . 'ORDER BY ul.title';
+
+        $parameters = ['user' => $this->getDoctrineReference(User::class, $userOrId)];
+        $query = $this->entityManager->createQuery($dql);
+        $query->setParameters($parameters);
+        $results = $query->getResult();
+        return $results;
+    }
+
+    /**
+     * Retrieve a batch of list objects corresponding to the provided IDs
+     *
+     * @param int[] $ids List ids.
+     *
+     * @return array
+     */
+    protected function getUserListsById(array $ids): array
```

Comment:
Right now this is a support method, but we might want to consider promoting it to a public method and adding it to the interface. We might also be able to eliminate the need for it by refactoring the query in getUserListsByTagAndId, so I'm leaving it protected for the moment to keep the interface as small as possible.